src/util/Combinatorics.java

```diff
@@ -0,0 +1,39 @@
+package util;
+
+import java.util.*;
+
+public class Combinatorics {
+
+    public static List<Set> allCombinationsOf(Set objects) {
```

Comment:
when you say, generic method, I may do that:

public static List<Set<Object>> allCombinationsOf(Set<? extends Object> objects) {
        List<Set<Object>> result = new ArrayList<Set<Object>>();
.....